### PR TITLE
Feat: cli: Remove verified data cap

### PR DESCRIPTION
**Pull request by geoff-vball**
## Related Issues
Closes Issue #7863 

## Proposed Changes
Added the following cli commands:
lotus filplus sign-remove-data-cap-proposal 
lotus-shed verifreg remove-verified-client-data-cap

## Additional Info
Tested on devnet. Works as expected.

## Checklist

Before you mark the PR ready for review, please make sure that:
- [ ] All commits have a clear commit message.
- [ ] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] CI is green

### CODE REVIEW

File: cli/filplus.go [hunk 1]
```diff
@@ -274,3 +277,105 @@ func checkNotary(ctx context.Context, api v0api.FullNode, vaddr address.Address)
...
+	Flags: []cli.Flag{
+		&cli.Int64Flag{
+			Name:     "id",
+			Usage:    "specify the id of the Remove Data Cap Proposal",
+			Required: false,
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		if cctx.Args().Len() != 3 {
+			return fmt.Errorf("must specify three arguments: verifier address, client address, and allowance to remove")
```
Comment (arajasek) [inline cli/filplus.go:293]:
```suggestion
			return fmt.Errorf("must specify three arguments: notary address, client address, and allowance to remove")
```

File: cmd/lotus-shed/verifreg.go [hunk 2]
```diff
@@ -409,3 +413,115 @@ var verifRegCheckVerifierCmd = &cli.Command{
 		return nil
 	},
 }
+
+var verifRegRemoveVerifiedClientDataCapCmd = &cli.Command{
+	Name:      "remove-verified-client-data-cap",
+	Usage:     "Remove data cap from verified client",
+	ArgsUsage: "<message sender> <verified client ID> <allowance to remove> <verifier 1 ID> <verifier 1 signature> <verifier 2 ID> <verifier 2 signature>",
```
Comment (arajasek) [inline cmd/lotus-shed/verifreg.go:420]:
```suggestion
	ArgsUsage: "<message sender> <client address> <allowance to remove> <verifier 1 address> <verifier 1 signature> <verifier 2 address> <verifier 2 signature>",
```

File: cmd/lotus-shed/verifreg.go [hunk 3]
```diff
@@ -409,3 +413,115 @@ var verifRegCheckVerifierCmd = &cli.Command{
...
+	Action: func(cctx *cli.Context) error {
+		if cctx.Args().Len() != 7 {
+			return fmt.Errorf("must specify seven arguments: sender, client, allowance to remove, verifier 1 ID, verifier 1 signature, verifier 2 ID, verifier 2 signature")
```
Comment (arajasek) [inline cmd/lotus-shed/verifreg.go:423]:
```suggestion
			return fmt.Errorf("must specify seven arguments: sender, client, allowance to remove, verifier 1 address, verifier 1 signature, verifier 2 address, verifier 2 signature")
```

File: cmd/lotus-shed/verifreg.go [hunk 4]
```diff
@@ -409,3 +413,115 @@ var verifRegCheckVerifierCmd = &cli.Command{
...
+		if err != nil {
+			return err
+		}
+
+		allowanceToRemove, err := types.BigFromString(cctx.Args().Get(2))
+		if err != nil {
+			return err
+		}
+
+		verifier1Addr, err := address.NewFromString(cctx.Args().Get(3))
```
Comment (arajasek) [inline cmd/lotus-shed/verifreg.go:450]: let's resolve all these things (so that users can specify the robust addresses if they want)

Comment (arajasek) [reply cmd/lotus-shed/verifreg.go:450]: (so the 2 verifiers and the clients)

File: cmd/lotus-shed/verifreg.go [hunk 5]
```diff
@@ -409,3 +413,115 @@ var verifRegCheckVerifierCmd = &cli.Command{
...
+			VerifierRequest2: verifreg.RemoveDataCapRequest{
+				Verifier:          verifier2Addr,
+				VerifierSignature: sig2,
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		vrk, err := api.StateVerifiedRegistryRootKey(ctx, types.EmptyTSK)
```
Comment (arajasek) [inline cmd/lotus-shed/verifreg.go:496]: let's `StateGetActor` this, load the multisig state, call `Signers()`, and  confirm that `sender` is one of the signers.

**Review by arajasek — COMMENTED** [review]
Various nits

**Comment by arajasek** [comment]
Needs make gen

File: cmd/lotus-shed/verifreg.go (hunk 4 above)
Comment (geoff-vball) [reply cmd/lotus-shed/verifreg.go:452]: Robust addresses will get resolved by the function that processes this message.

File: chain/actors/builtin/verifreg/util.go [hunk 6]
```diff
@@ -50,3 +51,28 @@ func forEachCap(store adt.Store, ver actors.Version, root rootFunc, cb func(addr
...
+	}
+	if client.Protocol() != address.ID {
+		return false, 0, xerrors.Errorf("can only look up ID addresses")
+	}
+	vh, err := root()
+	if err != nil {
+		return false, 0, xerrors.Errorf("loading verifreg: %w", err)
+	}
+	if vh == nil {
+		return false, 0, xerrors.Errorf("remove data cap proposal hamt not found. you are probably using an incompatible version of actors")
```
Comment (arajasek) [inline chain/actors/builtin/verifreg/util.go:67]:
```suggestion
		return false, 0, xerrors.Errorf("remove data cap proposal hamt not found (only exists from v7 and after)")
```

**Review by arajasek — APPROVED** [review]

**Pull request merged**